app/src/main/java/ca/llamabagel/transpo/data/SearchRepository.kt

```diff
@@ -4,29 +4,80 @@
-        val stops = database.stopQueries.getStopsByName("$query*").executeAsList()
-        val routes = database.routeQueries.getRoutes("$query%").executeAsList()
+object OttawaBoundaries {
+    const val MIN_LAT = 45.203039
+    const val MIN_LNG = -76.227729
+    const val MAX_LAT = 45.513359
+    const val MAX_LNG = -75.351405
+}
+
+class SearchRepository @Inject constructor(private val database: TransitDatabase, private val strings: StringsGen) {
+
+    suspend fun getSearchResults(query: String): List<SearchResult> = withContext(Dispatchers.IO) {
+        if (query.isEmpty()) return@withContext emptyList<SearchResult>()
+
+        val stops = getStops(query)
+        val routes = getRoutes(query)
+        val places = getPlaces(query)
 
         val searchResults = mutableListOf<SearchResult>()
 
         if (routes.isNotEmpty()) {
-            searchResults.add(SearchResult.CategoryHeader("Routes"))
-            searchResults.addAll(routes.map { SearchResult.RouteItem("Name", it.short_name, it.type.toString()) })
+            searchResults.add(SearchResult.CategoryHeader(strings.get(R.string.search_category_routes)))
+            searchResults.addAll(routes)
         }
 
         if (stops.isNotEmpty()) {
-            searchResults.add(SearchResult.CategoryHeader("Stops"))
-            searchResults.addAll(stops.map { SearchResult.StopItem(it.name, "• ${it.code}", "No upcoming trips") })
+            searchResults.add(SearchResult.CategoryHeader(strings.get(R.string.search_category_stops)))
+            searchResults.addAll(stops)
+        }
+
+        if (places.isNotEmpty()) {
+            searchResults.add(SearchResult.CategoryHeader(strings.get(R.string.search_category_places)))
+            searchResults.addAll(places)
         }
 
         searchResults
     }
+
+    private suspend fun getStops(query: String): List<SearchResult.StopItem> = withContext(Dispatchers.IO) {
+        database.stopQueries
+            .getStopsByName("$query*")
+            .executeAsList()
+            .map { SearchResult.StopItem(it.name, "• ${it.code}", strings.get(R.string.search_stop_no_trips)) }
+    }
+
+    private suspend fun getRoutes(query: String): List<SearchResult.RouteItem> = withContext(Dispatchers.IO) {
+        database.routeQueries
+            .getRoutes("$query%")
+            .executeAsList()
+            .map { SearchResult.RouteItem("Name", it.short_name, it.type.toString()) } // TODO: update name parameter
+    }
+
+    private suspend fun getPlaces(query: String): List<SearchResult.PlaceItem> = withContext(Dispatchers.IO) {
+        MapboxGeocoding.builder()
+            .accessToken(MAPBOX_KEY)
+            .query(query)
+            .bbox(MIN_LNG, MIN_LAT, MAX_LNG, MAX_LAT)
+            .build()
+            .executeCall()
+            .body()
+            ?.features()
+            ?.map { SearchResult.PlaceItem(it.placeName().orEmpty(), it.text().orEmpty()) }
```

Comment:
```suggestion
            ?.map { feature -> SearchResult.PlaceItem(feature.placeName().orEmpty(), feature.text().orEmpty()) }
```
Add explicit lambda parameter name here

Reply:
Any particular reason for this? It's not overshadowed or anything.

Reply:
I think it's more readable this way. it explicitly states what's being dealt with in the closure since it's a bit too confusing to infer at the bottom of a long chain of function calls.